lib/block_view_stake.go

```diff
@@ -1955,7 +1955,7 @@ func (bav *UtxoView) IsValidUnlockStakeMetadata(transactorPkBytes []byte, metada
 	}
 
 	// Validate EndEpochNumber + StakeLockupEpochDuration <= CurrentEpochNumber.
-	if metadata.EndEpochNumber+bav.Params.StakeLockupEpochDuration > currentEpochNumber {
+	if metadata.EndEpochNumber+bav.GetStakeLockupEpochDuration(0) > currentEpochNumber {
```

Comment:
Should we be fetching the StakeLockupEpochDuration based on a snapshot?

Reply:
Yes. We should be fetching those based on a snapshot. For now, I'm just keeping the existing behavior. I'll update exactly which snapshot we should fetch once I know it. That'll just be swapping out the `0` for the correct EpochNumber relative to the CurrentEpochNumber.

Reply:
Yes. We should be fetching those based on a snapshot. For now, I'm just keeping the existing behavior. I'll update exactly which snapshot we should fetch once I know it. That'll just be swapping out the `0` for the correct EpochNumber relative to the CurrentEpochNumber.